projects/core/components/hosted-dropdown/hosted-dropdown.component.ts

```diff
@@ -42,6 +43,10 @@ export interface TuiHostedDropdownContext
     close(): void;
 }
 
+function activeZoneFilter(this: TuiHostedDropdownComponent): boolean {
+    return this.isLastOpenedDropdown && this.activeZone.contains(this.wrapperElement);
```

Comment:
This is not what we need to be testing here. We need to test if the key pressed is escape and that the dropdown is open (`this.open`).

Reply:
<img width="826" alt="image" src="https://user-images.githubusercontent.com/12021443/222660647-cdbf0ac3-6b4a-4607-88b2-4f8acaad16d6.png">

And so, we have 2 open dropdown box. Expected behavior after the escape only the last one will be closed and so that it is:

<img width="801" alt="image" src="https://user-images.githubusercontent.com/12021443/222660850-ba6d4168-67ca-4118-9a86-ea8cbb9f225f.png">